Improvements for related agenda items

emanuelschuetze opened the pull request:
- Added QuickEdit mode for related agenda items
- support hidden items
- Added ng-dialog for modal create/update dialogs of customslides
- use generic links for list of speakers, edit, delete

Some template style issues are still open. I'll fix it with my next pull request which brings up a new base template. So please merge this asap. Thanks!

normanjaeckel commented:
Looks ok.

ostcar commented on openslides/agenda/static/js/agenda/site.js:
```diff
@@ -134,8 +191,19 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
+                            Assignment.destroy(item.content_object.id);
+                            break;
+                    }
+                }
```
I don't like this code, because the agenda app should be independent from Motion and Assignment. But it is not needed anyway, because the Customslide-, Motion- or Assignmentobject is deleted from the server when the related item is deleted.

emanuelschuetze replied:
I changed it back to `Agenda.destroy(item.id)`.
But now, this code does not allowed me to delete an agenda item (Customslide, Motion, Assignment)! I get a `403 Forbidden` for the DELETE request (e.g. `DELETE http://localhost:8000/rest/agenda/item/10/`).

ostcar commented on openslides/users/static/js/users/site.js:
```diff
@@ -620,6 +621,59 @@ angular.module('OpenSlidesApp.users.site', ['OpenSlidesApp.users'])
+            type: 'success',
+            msg: gettext("Installation was successfully.<br>" +
+                "Use <strong>admin</strong> and <strong>admin</strong> for first login." +
+                "<p>Important: Please change your password! ")
```
Proposal: Try to remove the html-tags from the string, so it is not part of the translation.

emanuelschuetze commented:
Rebased on top of master. Added review commit. Still a open question about not working DELETE request for agenda items.

normanjaeckel commented:
It was a design decision not to allow DELETE requests for agenda items because they should be fully related to the obejct behind it (motion, custom slide etc). Deleting an agenda item would break the datastructure at this point and open some more questions to handle.

emanuelschuetze commented:
@ostcar So, how can we solve quickly the DELETE topic today? Should I add my old code (with Motions,Assignments etc.) again?

emanuelschuetze commented:
I just added two template fixes.

ostcar commented:
@emanuelschuetze I have no problem with your code for the customslide. You can add this. But the agenda list view should not have buttons to delete a motion or assignment. My suggestion is, that you check in the template if the item is a custom slide item and only then show the delete button.

emanuelschuetze commented:
Rebased on top of master.
Added 'Review2' commit which deletes only customslides (not motions/assignments) in ItemListCtrl.
One last thing: How can I project related items (not only the agenda items)? ìtem.project()`is showing the title of agenda item only. I search something like `Customslide.find(item.content_object.id).project();`which does not work...
@oskar?

emanuelschuetze commented:
Project function now works with 'review 3' commit.
@ostcar Thanks!
So please do a last review. After that I will rebase.

emanuelschuetze commented:
If you have a quick solution for this pull request:
`<div ng-bind-html="customslide.title"></div>`shows nothing on customslide slide template. I can't see why. So I comment it out in review3 for this pr and I'll fix it later.

ostcar commented:
ok. Please rebase and merge

emanuelschuetze commented:
All done. Rebased and fixed also the ng-bind-html by moving ngSanitize to core/base.js.
So ready for merge!
